tasks/debian.yml

```diff
@@ -24,13 +24,29 @@
   retries: 5
   delay: 5
 
+- name: Temp file for check mode
+  set_fact:
+    __temp_file_for_check_mode: "/tmp/galaxyproject_postgresql_get_installed_version_trick"
+
+- name: Create temp file for check_mode
+  ansible.builtin.copy:
+    content: ""
+    dest: "{{ __temp_file_for_check_mode }}"
+    force: yes
+    mode: '0777'
+  check_mode: no
+
 - name: Get installed version
-  command: dpkg-query -f ${Version;3} --show postgresql
+  shell: "dpkg-query -f '${Version;3}' --show postgresql"
   when: postgresql_version is not defined
   register: __postgresql_version_query_result
   changed_when: false
+  check_mode: no
+  args:
+    # Using this in order to enforce the task to be executed in check mode
+    removes: "{{ __temp_file_for_check_mode }}"
```

Comment:
since we're shelling out to dpkg-query anyway, could we use something like the [package facts module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_facts_module.html) instead that might be natively check-safe? @natefoo ?

Reply:
Do you think it is ok even though it has the following `requirements`?

> The below requirements are needed on the host that executes this module.
> 
>     For ‘portage’ support it requires the qlist utility, which is part of ‘app-portage/portage-utils’.
> 
>     For Debian-based systems python-apt package must be installed on targeted hosts.

Reply:
Oh, that's not ideal.

Reply:
Yeah, I thought so....